81-IDevID_and_IAK_support.md

```diff
@@ -0,0 +1,302 @@
+6. Encode the necessary data and send it to the registrar service (modify the method do_register_agent to support the new parameters regarding IDevID, LDevID, IAK and LAK keys). 
+7. Receive the challenge to prove that is actually has the IDevID, LDevID, IAK and LAK keys. 
+8. Sends challenge response back to the registrar service.
+9. The agent gets its UUID in case of a successful registering process.
+
+### Registration from the registrar service perspective when interacting with the agent
+1. The main endpoints here are do_register_agent() and do_activate_agent() invoked by the agent.
+2. Decode data and parse certificates. This implies modifying the method doRegisterAgent to support new parameters regarding IAK and IDevID. This method exercises a POST to /v{api_version}/agents/{agent_id} at registrar_common.py do_POST method.  
+3. Verify IDevID certificate chain of trust + IAK certificate chain of trust
+4. Create a challenge to send to the agent for prove the possession of the private keys. It's important to mention that the challenge will change from using Make/ActivateCredential to use TPM2_Certify.
+4. Create a challenge to send to the agent for prove the possession of the private keys. 
+5. Receive the challenge response and process
+6. Generate the certificates for LAK and LDevID keys with the owner/company's CAs. 
+6. Commit data to the local Keylime database, the necessary fields regarding keys and certificates to be included there.  
+7. Activate the agent. This implies modifying the method do_activate_agent at keylime\keylime\registrar_client.py to support new parameters regarding IAK and IDevID. This method exercises a PUT to /v{api_version}/agents/{agent_id}/activate at registrar_common.py do_PUT method.  
+
+### Attestation from the verifier perspective after the registration process
+The work from the verifier perspective can keep using the ephemeral AK or switch between all the possible scenarios mentioned. 
+This is being thought as a multi-step plan and these scenarios will be implemented gradually. 
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+TBD
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this enhancement
+-->
+TBD
```

Comment:
Downgrade: don't use that feature 
Upgrade: use new API version to use this feature.